### PR TITLE
Further optimize (geohashes-intersecting) #25

### PR DESCRIPTION
**Pull request by chen-factual**
* 1.7x speed up on splitting test US polygon into L3 geohashes

### CODE REVIEW

File: src/geo/geohash.clj [hunk 1]
```diff
@@ -194,22 +194,48 @@
 
 (defn- queue [] clojure.lang.PersistentQueue/EMPTY)
 
+(defn- geohashes-intersecting-matches!
+  "Given geohash and shape-relation, modify the matches list as necessary
+   :disjoint    no matches
+   :intersects  current geohash matches, but can't extend assertion to subdivisions
+   :within      same as within
```
Comment (worace) [inline src/geo/geohash.clj]: does this mean `same as :contains`?

File: src/geo/geohash.clj [hunk 2]
```diff
@@ -194,22 +194,48 @@
...
+   :contains    all subdivisions at all valid levels match"
+  [min-level max-level gh shape-relation matches]
+  (case shape-relation
+   ;; gh fully in shape. Add subdivisions of all valid
+   ;; levels into matches
+    :contains
+   (let [start-level (max (significant-bits gh) min-level)
```
Comment (worace) [inline src/geo/geohash.clj]: Could this be pulled out into a function...maybe something like `(defn children-at-levels [gh min-level max-level] ...)`?

File: src/geo/geohash.clj (hunk 1 above)
Comment (chen-factual) [reply src/geo/geohash.clj]: I actually meant `same as :intersects`, as in the treatment is the same. Sharp eye, I'll update.

**Comment by chen-factual** [comment]
Updated, added test for split out function which actually caught a bug.

**Pull request merged**